sympy/calculus/util.py

```diff
@@ -373,6 +375,16 @@ def periodicity(f, symbol, check=False):
 
         period = _periodicity(g.args, symbol)
 
+    elif isinstance(f, Mod):
+        a, n = f.args
+
+        if a == symbol:
+            period = n
+        elif isinstance(a, TrigonometricFunction):
+            period = periodicity(a, symbol)
+        elif degree(a, symbol) == 1: #check if 'f' is linear in 'symbol'
+            period = Abs(n/a.coeff(symbol))
```

Comment:
this is not robust enough:

```
>>> (sqrt(2)*(x+1)-x).coeff(x)
-1 # instead of sqrt(2) - 1
>>> (sqrt(2)*(x+1)-x).diff(x)
-1 + sqrt(2)
```

I recommend that you use `Abs(n/a.diff(symbol))`